File: Leetcode/dfs_recursion_backtracking/118_Pascal's Triangle/Solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        if(numRows==0)
            return new LinkedList<List<Integer>>();
        List<List<Integer>> res = generate(numRows-1);
        LinkedList<Integer> temp = new LinkedList<>();
        temp.add(1);
        if(numRows==1){
            res.add(temp);
        }else{
            List<Integer> last = res.get(res.size()-1);
            for(int i = 0; i<last.size()-1; ++i){
                temp.add(last.get(i)+last.get(i+1));
            }
            temp.add(1);
            res.add(temp);
        }
        return res;
    }
}

File: Leetcode/dfs_recursion_backtracking/118_Pascal's Triangle/Solution.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if numRows==0:
            return []
        res = [[1]]
        for i in range(1, numRows):
            last = res[-1]
            res += [[1]+[last[i]+last[i+1] for i in range(len(last)-1)]+[1]]
        return res
